tracer/src/Datadog.Trace/PDBs/DatadogMetadataReader.cs

```diff
@@ -76,7 +76,7 @@ private DatadogMetadataReader(PEReader peReader, MetadataReader metadataReader,
             if (peReader.TryOpenAssociatedPortablePdb(assembly.Location, File.OpenRead, out var metadataReaderProvider, out var pdbPath))
             {
                 pdbReader = metadataReaderProvider!.GetMetadataReader(MetadataReaderOptions.Default, MetadataStringDecoder.DefaultUTF8);
-                return new DatadogMetadataReader(peReader, metadataReader, pdbReader, pdbPath, null, null);
+                return new DatadogMetadataReader(peReader, metadataReader, pdbReader, pdbPath ?? assembly.Location, null, null);
```

Comment:
It's ensure that `pdbReader` isn't null but not `pdbPath`. For the other part, it's just to easier the usage later when trying to use `PdbFullPath`. I've added a comment on this property that says: "Return the pdb path if exists, otherwise the assembly location". Do you prefer to do something else?

9f23fdedaa2a7645d2ae2d2edbaf79936c9e57ef